include/roboteam_ai/stp/StpInfo.h

```diff
@@ -69,11 +69,11 @@ namespace rtt::ai::stp {
          */
         world_new::view::RobotView passRobot;
 
+        Vector2 getPosition() const { return position; };
+        void setPosition(Vector2 position) { this->position = position; };
 
-        /**
-         * Position, could be position of robot, or whatever position the skill needs, maybe target position
-         */
-        Vector2 position;
+        std::optional<Field> getField() const { return field; };
+        void setField(std::optional<Field> field) { this->field = field; };
```

Comment:
I think this function should not take an optional as parameter, since you should actually have a field when you call `setField`, not maybe have a field.